x/protorev/keeper/routes_test.go

```diff
@@ -0,0 +1,249 @@
+				}
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestBuildAtomRoute() {
+	cases := []struct {
+		description   string
+		swapIn        string
+		swapOut       string
+		poolId        uint64
+		expectedRoute []TestRoute
+		hasRoute      bool
+	}{
+		{
+			description:   "Route exists for swap in Osmo and swap out Akash",
+			swapIn:        types.OsmosisDenomination,
+			swapOut:       "akash",
+			poolId:        7,
+			expectedRoute: []TestRoute{{1, types.AtomDenomination, "akash"}, {7, "akash", types.OsmosisDenomination}, {25, types.OsmosisDenomination, types.AtomDenomination}},
+			hasRoute:      true,
+		},
+		{
+			description:   "Route exists for swap in Akash and swap out Osmo",
+			swapIn:        "akash",
+			swapOut:       types.OsmosisDenomination,
+			poolId:        7,
+			expectedRoute: []TestRoute{{25, types.AtomDenomination, types.OsmosisDenomination}, {7, types.OsmosisDenomination, "akash"}, {1, "akash", types.AtomDenomination}},
+			hasRoute:      true,
+		},
+		{
+			description:   "Route exists for swap in Terra and swap out Osmo (no mapping pool)",
+			swapIn:        "terra",
+			swapOut:       types.OsmosisDenomination,
+			poolId:        7,
+			expectedRoute: []TestRoute{},
+			hasRoute:      false,
+		},
+	}
+
+	for _, tc := range cases {
+		suite.Run(tc.description, func() {
+			route, err := suite.App.ProtoRevKeeper.BuildAtomRoute(suite.Ctx, tc.swapIn, tc.swapOut, tc.poolId)
+
+			if tc.hasRoute {
+				suite.Require().NoError(err)
+				suite.Require().Equal(len(tc.expectedRoute), len(route))
+
+				for index, trade := range tc.expectedRoute {
+					suite.Require().Equal(trade.PoolId, route[index].Pool.GetId())
+					suite.Require().Equal(trade.InputDenom, route[index].InputDenom)
+					suite.Require().Equal(trade.OutputDenom, route[index].OutputDenom)
+				}
+			} else {
+				suite.Require().Error(err)
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestBuildOsmoRoute() {
+	cases := []struct {
+		description   string
+		swapIn        string
+		swapOut       string
+		poolId        uint64
+		expectedRoute []TestRoute
+		hasRoute      bool
+	}{
+		{
+			description:   "Route exists for swap in Atom and swap out Akash",
+			swapIn:        types.AtomDenomination,
+			swapOut:       "akash",
+			poolId:        1,
+			expectedRoute: []TestRoute{{7, types.OsmosisDenomination, "akash"}, {1, "akash", types.AtomDenomination}, {25, types.AtomDenomination, types.OsmosisDenomination}},
+			hasRoute:      true,
+		},
+		{
+			description:   "Route exists for swap in Akash and swap out Atom",
+			swapIn:        "akash",
+			swapOut:       types.AtomDenomination,
+			poolId:        1,
+			expectedRoute: []TestRoute{{25, types.OsmosisDenomination, types.AtomDenomination}, {1, types.AtomDenomination, "akash"}, {7, "akash", types.OsmosisDenomination}},
+			hasRoute:      true,
+		},
+		{
+			description:   "Route exists for swap in Terra and swap out Atom (no mapping pool)",
```

Comment:
Shouldn't this description be "Route doesnot exist for swap in Terra and swap out Atom because the pool doesnot exist"?

Reply:
yea you are correct. my comment is a little confusing without context. i'll switch it up to what you suggested.